### PR TITLE
[Platform] Update L10n Linter strategy to allow retry

### PR DESCRIPTION
**Pull request by AliSoftware**
This PR takes advantage of the new mode of the `ios_lint_localizations` action [recently added in release-toolkit 0.12.0](https://github.com/wordpress-mobile/release-toolkit/pull/182).

[Internal ref: paaHJt-1Fn-p2#comment-3524]

## What changed?

After this PR, if a localization violation is found, we will now be given the possibility to fix the violation locally and retry until we get it right – as opposed to waiting for it to be fixed in GlotPress.

This new strategy applies to **both** the `new_beta_release` and `finalize_release`.

_Note: Ideally this is still better to solve the violations at their source in GlotPress first then re-run the lanes from scratch. But it's also not always possible to wait for someone to be available to solve it in GlotPress for us; so this new behavior gives us an escape hatch when we can't wait for a fix of the copy in GP. We should still ping someone to fix the issue in GP in all cases._

## Target branch

This PR intentionally targets the `release/16.4` branch and not `develop` because we're hoping to already use this new behavior for the upcoming release of 16.4, instead of waiting for the next code freeze to include it.

## To Test

The test steps are similar to the ones from https://github.com/wordpress-mobile/release-toolkit/pull/182:

 - Introduce a violation (incorrect `%d`/`%@`/… placeholders) in one of the `.strings` file
 - Make a temporary edit to the `finalize_release` lane to test the PR:
   - Comment the `ios_update_metadata()` call (to avoid your temporary violation from step above to be overwritten)
   - Replace the `ios_bump_version_beta()` and `ios_tag_build()` calls with something like `UI.message("And now new version is tagged")` (to be able to see if/when the next steps after `ios_lint_localizations` will be called as expected)
 - Run the modified lane: `bundle exec fastlane finalize_release` and ensure it finds the violations you introduced on purpose earlier.
 - Fix the violation in the local `.strings` file, then choose "Retry".
 - Check that the linter pass on this second try after violation has been fixed.

## Other: Security Fix

This PR will also incidentally close [this Dependabot security alert](https://github.com/wordpress-mobile/WordPress-iOS/security/dependabot/Gemfile.lock/nokogiri/open) from our repo – which was a dependency of `release-toolkit`, which has been fixed in its version 0.12.0.

### CODE REVIEW

**Comment by AliSoftware** [comment]
Note: CI failure is unrelated and due to a [broken test that was recently fixed in develop](https://github.com/wordpress-mobile/WordPress-iOS/pull/15577); @emilylaguna is already planning to cherry pick the fix into the release branch at some point 🙇 .

**Comment by AliSoftware** [comment]
The CI failure has now been fixed by https://github.com/wordpress-mobile/WordPress-iOS/pull/15600, so just rebased this PR on top of release/16.4

**Comment by AliSoftware** [comment]
@jkmassel Let me know once you did the code freeze for 16.5 and I'll rebase this PR on top of `release/16.5` once the branch exists.

**Comment by jkmassel** [comment]
👋 We're freezing WPiOS `16.5` today, so this PR is being bumped to `16.6`. 

If you need it to be part of the `16.5` release, please DM me – I'll be happy to cut a new beta release once this lands!

**Comment by jkmassel** [comment]
@AliSoftware – ready for a rebase when you're ready!

**Review by loremattei — APPROVED** [review]
LGTM!
Thanks for this improvement :-)

**Comment by AliSoftware** [comment]
@jkmassel rebased on top of `release/16.5` and changed the PR's target branch.
I'll let you merge the PR once you give it a last check post-rebase and once CI is green.

**Review by jkmassel — APPROVED** [review]

**Comment by jkmassel** [comment]
Thanks @AliSoftware!!

**Pull request merged**
